cfme/tests/services/test_provision_stack.py

```diff
@@ -181,35 +205,45 @@ def create_template(setup_provider, provider, dialog):
 
 
 def prepare_stack_data(provider, provisioning):
-    stackname = "test" + fauxfactory.gen_alphanumeric()
+    stack_name = "test" + fauxfactory.gen_alphanumeric()
+    vm_name = stack_name
     if provider.type == "azure":
-        vm_name = "test" + fauxfactory.gen_alphanumeric()
         vm_user, vm_password, vm_size, resource_group,\
             user_image, os_type, mode = map(provisioning.get,
          ('vm_user', 'vm_password', 'vm_size', 'resource_group',
         'user_image', 'os_type', 'mode'))
 
         stack_data = {
-            'stack_name': stackname,
+            'stack_name': stack_name,
             'vm_name': vm_name,
             'resource_group': resource_group,
             'mode': mode,
             'vm_user': vm_user,
             'vm_password': vm_password,
-            'vm_size': vm_size
+            'vm_size': vm_size,
+        }
+        return stack_data
+    elif provider.type == 'ec2':
+        guest_keypair, ssh_location = map(provisioning.get, ('guest_keypair', 'ssh_location'))
+        stack_data = {
+            'stack_name': stack_name,
+            'key_name': guest_keypair,
+            'vm_name': vm_name,
+            'ssh_location': ssh_location
         }
         return stack_data
     else:
-        stack_data = {'stack_name': stackname}
+        stack_data = {'stack_name': stack_name}
         return stack_data
 
 
-def test_provision_stack(provider, provisioning, create_template, catalog, request):
+def test_provision_stack_x(provider, provisioning, create_template, catalog, request):
```

Comment:
I did this because the test name is the same as the file name, so you can't call a single test using -k in this circumstance.  So for testing the WIP, I changed the name so I could isolate the single test.  Always planned to change it back.

Note to JT:    DON"T FORGET TO CHANGE THIS BACK.